contracts/SpokePool.sol

```diff
@@ -607,14 +614,18 @@ abstract contract SpokePool is
             IERC20Upgradeable(inputToken).safeTransferFrom(msg.sender, address(this), inputAmount);
         }
 
+        // Add the last known deposit count to the depositor's ID to ensure that all deposit IDs are unique including
+        // all past deposits which used numberOfDeposits as a counter.
+        uint32 accountDepositCount = depositCount(depositor);
```

Comment:
An alternative cheaper on gas implementation is to add a deposit entry point where the caller can set `depositId`. This way the gasless deposit receiver like the filler can pre-fill in the depositId using the Permit2 nonce, for example. This function would come with the caveat that the caller shoudl be sure that their deposit ID has not been used yet otherwise the deposit will be unfillable and their deposit will expired